src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+    return new BigDecimal(numProjectLibraries).divide(new BigDecimal(numLaneLibraries), 1,  RoundingMode.HALF_UP);
+  }
+  
+  private String getInstrumentModel(Integer instrumentId, Map<Integer, InstrumentDto> instruments, Map<Integer, InstrumentModelDto> models) {
+    return models.get(
+        instruments.get(instrumentId).getModelId())
+      .getName();
+  }
+  
+  private String getInstrumentName(Integer instrumentId, Map<Integer, InstrumentDto> instrumentsById) {
+    InstrumentDto instrument = instrumentsById.get(instrumentId);
+    return instrument == null ? "Unknown" : instrument.getName();
+  }
+  
+  private String getSummaryKey(DetailedObject row) {
+    return row.getProject() + ":" + row.getInstrumentModel();
+  }
+  
+  private Map<String, BigDecimal> makeNewSummaryLanes(String laneType, BigDecimal laneCount) {
+    Map<String, BigDecimal> lanes = new HashMap<String, BigDecimal>();
+    lanes.put(DNA_LANE, BigDecimal.ZERO);
+    lanes.put(RNA_LANE, BigDecimal.ZERO);
+    lanes.put(MIXED_LANE, BigDecimal.ZERO);
+    addSummaryLanes(lanes, laneType, laneCount);
+    return lanes;
+  }
+  
+  private void addSummaryLanes(Map<String, BigDecimal> lanes, String laneType, BigDecimal laneCount) {
+    lanes.merge(laneType, laneCount, BigDecimal::add);
+  }
+  
+  static final List<Map<String, Map<String, BigDecimal>>> listifySummary(Map<String, Map<String, BigDecimal>> summary) {
+    // need to convert it to a list, because getRow() takes an index and the treemap doesn't yet have one of those
+    List<Map<String, Map<String, BigDecimal>>> regrettable = new ArrayList<Map<String, Map<String, BigDecimal>>>();
+    
+    for (Map.Entry<String, Map<String, BigDecimal>> entry : summary.entrySet()) {
+      Map<String, Map<String, BigDecimal>> oneProjModel = new HashMap<>();
+      oneProjModel.put(entry.getKey(), entry.getValue());
+      regrettable.add(oneProjModel);
+    }
+    return regrettable;
+  }
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    return Arrays.asList(
+        new ColumnDefinition("Project"),
+        new ColumnDefinition("Instrument Model"),
+        new ColumnDefinition(DNA_LANE),
+        new ColumnDefinition(RNA_LANE),
+        new ColumnDefinition(MIXED_LANE),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition("")
+    );
+  }
+  
+  private List<String> getDetailedHeadings() {
+    return Arrays.asList(
+        "Project",
+        "Instrument Name",
+        "Instrument Model",
+        "Run Status",
+        "Run End Date",
+        "Run Name",
+        "Lane",
+        DNA_LANE,
+        RNA_LANE,
+        MIXED_LANE
+    );
+  }
+
+  @Override
+  protected int getRowCount() {
+    return completed.getDetailedData().size()
+        + (completed.getDetailedData().size() > 0 ? 2 : 0)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)